File: src/main/java/spixie/Multiplier.java
package spixie;

import javafx.scene.control.Label;
import javafx.scene.layout.VBox;

public class Multiplier extends VBox implements Element {
    final public Value radius = new Value(30.0 ,1.0, "Radius");
    final public Value phase = new Value(0 ,0.001, "Phase");
    final public Value size = new Value(15,1.0, "Size");
    final public Value count = new Value(5,0.1,"Count");
    public Multiplier() {
        super();
        getChildren().addAll(new Label("spixie.Multiplier"));
        getChildren().addAll(radius, phase, size, count);
    }

    public Value.Item[] getValues(){
        return new Value.Item[]{radius.item(), phase.item(), size.item(), count.item()};
    }

    @Override
    public void addGraph(Value outValue) {
        Graph graph = new Graph(this, outValue);
        getChildren().addAll(graph);
    }
}

File: src/main/java/spixie/Multiplier.kt
package spixie

import javafx.scene.control.Label
import javafx.scene.layout.VBox

class Multiplier : VBox(), Element {
    val radius = Value(30.0, 1.0, "Radius")
    val phase = Value(0.0, 0.001, "Phase")
    val size = Value(15.0, 1.0, "Size")
    val count = Value(5.0, 0.1, "Count")

    init {
        children.addAll(Label("Multiplier"))
        children.addAll(radius, phase, size, count)
    }

    override val values: Array<Value.Item>
        get() = arrayOf(radius.item(), phase.item(), size.item(), count.item())

    override fun addGraph(outValue: Value) {
        val graph = Graph(this, outValue)
        children.addAll(graph)
    }
}
